tests/Tests.AzureAppConfiguration/KeyVaultReferenceTests.cs

```diff
@@ -367,5 +367,122 @@ public void DoesNotThrowKeyVaultExceptionWhenProviderIsOptional()
             }, optional: true)
             .Build();
         }
+
+        [Fact]
+        public void CallsSecretResolverCallbackWhenNoMatchingSecretClientIsFound()
+        {
+            var mockResponse = new Mock<Response>();
+            var mockClient = new Mock<ConfigurationClient>(MockBehavior.Strict, TestHelpers.CreateMockEndpointString());
+            mockClient.Setup(c => c.GetConfigurationSettingsAsync(It.IsAny<SettingSelector>(), It.IsAny<CancellationToken>()))
+                .Returns(new MockAsyncPageable(new List<ConfigurationSetting> { _kv }));
+
+            IConfiguration config = new ConfigurationBuilder()
+                .AddAzureAppConfiguration(options =>
+                {
+                    options.Client = mockClient.Object;
+                    options.ConfigureKeyVault(kv =>
+                    {
+                        kv.SetSecretResolver((secretUri) =>
+                        {
+                            return new ValueTask<string>(secretUri.ToString());
+                        });
+                    });
+                })
+                .Build();
+
+            Assert.Equal("https://keyvault-theclassics.vault.azure.net/secrets/TheTrialSecret", config["TestKey1"]);
+        }
+
+        [Fact]
+        public void ThrowsWhenBothDefaultCredentialAndSecretResolverCallbackAreSet()
+        {
+            var mockResponse = new Mock<Response>();
+            var mockClient = new Mock<ConfigurationClient>(MockBehavior.Strict, TestHelpers.CreateMockEndpointString());
+            mockClient.Setup(c => c.GetConfigurationSettingsAsync(It.IsAny<SettingSelector>(), It.IsAny<CancellationToken>()))
+                .Returns(new MockAsyncPageable(new List<ConfigurationSetting> { _kv }));
+
+            ArgumentException ex = Assert.Throws<ArgumentException>(() =>
+            {
+                new ConfigurationBuilder().AddAzureAppConfiguration(options =>
+                {
+                    options.Client = mockClient.Object;
+                    options.ConfigureKeyVault(kv =>
+                    {
+                        kv.SetSecretResolver((secretUri) =>
+                        {
+                            return new ValueTask<string>(secretUri.ToString());
+                        });
+
+                        kv.SetCredential(new DefaultAzureCredential());
+                    });
+                })
+                .Build();
+            });
+
+            Assert.NotNull(ex.InnerException);
+            Assert.IsType<InvalidOperationException>(ex.InnerException);
```

Comment:
>Assert.IsType<InvalidOperationException>(ex.InnerException); [](start = 12, length = 60)

Why would it throw `InvalidOperationException` as inner exception instead of top-level exception?

Reply:
The `Build()` method wraps `InvalidOperationException` as `ArgumentException` here: https://github.com/Azure/AppConfiguration-DotnetProvider/blob/96abfe3af854bde6d92860e74ab59866c1df03ca/src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationSource.cs#L55-L61

Reply:
Do we understand why? It seems odd if the user receives an `ArgumentException` when both default credentials and secret resolver are specified.

Reply:
`ConfigureKeyVault` (or any viable code path where we can throw this exception) is called inside the `ConfigurationBuilder.Build()` method. But I am not sure why we specifically catch `InvalidOperationException` here and throw `ArgumentException` instead.

Reply:
I think we should re-assess if we can let the `InvalidOperationException` bubble up. Since this would be a breaking change, we can skip it for now, but its something we can re-visit in the future.